reconcile/utils/oc.py

```diff
@@ -725,15 +722,30 @@ def wait_for_job_running(self, namespace, name):
         if not ready_pods:
             raise JobNotRunningError(name)
 
-    def job_logs(self, namespace, name, follow, output):
-        self.wait_for_job_running(namespace, name)
+    def job_logs(
+        self,
+        namespace,
+        name,
+        follow,
+        output,
+        wait_for_job_running=True,
+        wait_for_logs_process=False,
+    ):
+        if wait_for_job_running:
+            self.wait_for_job_running(namespace, name)
+
         cmd = ["logs", "--all-containers=true", "-n", namespace, f"job/{name}"]
         if follow:
             cmd.append("-f")
         # pylint: disable=consider-using-with
-        output_file = open(os.path.join(output, name), "w", encoding="locale")
+        if isinstance(output, TextIOWrapper):
+            output_file = output
+        else:
+            output_file = open(os.path.join(output, name), "w", encoding="locale")
         # collect logs to file async
-        Popen(self.oc_base_cmd + cmd, stdout=output_file)
+        p = Popen(self.oc_base_cmd + cmd, stdout=output_file)
+        if wait_for_logs_process:
+            p.wait()
```

Comment:
process is not captured if `wait_for_logs_process` is `False`, main thread may finish before subprocess, maybe unexpected leak. Better to collect them using `subprocess.run`. To avoid block main thread, can start this in a new thread/pool. See [APPSRE-7403](https://issues.redhat.com/browse/APPSRE-7403)

Reply:
I'm just adding the option to get the logs to the stdout. I was unaware of that ticket but would prefer not to wait before this is fixed to use this feature. This can not be deprecated as is used by the JobController. 

Is it ok for you to add this change and tackle the refactor in APPSRE-7403?

Reply:
Then can we leave old code not changed but use `subprocess.run` for the new feature?

```python
args = self.oc_base_cmd + cmd
if wait_for_logs_process:
  subprocess.run(args, stdout=output_file)
else:
  # collect logs to file async
  Popen(args, stdout=output_file)
```

Reply:
Done.